### ISSUE TITLE
Failed to build native extensions

### ISSUE THREAD

**Issue by coen-hyde** [sparklemotion/nokogiri#1454]
Hi, 

I just upgraded nokogiri and now the gem fails to install. I narrowed down the breaking change between 1.6.7.rc3 and 1.6.7.rc4. This is on OSX 10.10.5.

Here is the log output from configure: 
`/Users/coen-hyde/.dotfiles/link/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/nokogiri-1.6.7.rc4/ext/nokogiri/tmp/x86_64-apple-darwin14.5.0/ports/libxslt/1.1.28/configure.log`

```
configure: WARNING: if you wanted to set the --build type, don't use --host.
    If a cross compiler is detected then cross compile mode will be used
checking build system type... ./config.guess: unable to guess system type

This script, last modified 2015-10-21, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
and
  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

If the version you run (./config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.

config.guess timestamp = 2015-10-21

uname -m = x86_64
uname -r = 14.5.0
uname -s = Darwin
uname -v = Darwin Kernel Version 14.5.0: Tue Sep  1 21:23:09 PDT 2015; root:xnu-2782.50.1~1/RELEASE_X86_64

/usr/bin/uname -p = i386
/bin/uname -X     =

hostinfo               = Mach kernel version:
         Darwin Kernel Version 14.5.0: Tue Sep  1 21:23:09 PDT 2015; root:xnu-2782.50.1~1/RELEASE_X86_64
Kernel configured for up to 4 processors.
2 processors are physically available.
4 processors are logically available.
Processor type: x86_64h (Intel x86-64h Haswell)
Processors active: 0 1 2 3
Primary memory available: 8.00 gigabytes
Default processor set: 289 tasks, 1397 threads, 4 processors
Load average: 2.97, Mach factor: 1.56
/bin/universe          =
/usr/bin/arch -k       =
/bin/arch              =
/usr/bin/oslevel       =
/usr/convex/getsysinfo =

UNAME_MACHINE = x86_64
UNAME_RELEASE = 14.5.0
UNAME_SYSTEM  = Darwin
UNAME_VERSION = Darwin Kernel Version 14.5.0: Tue Sep  1 21:23:09 PDT 2015; root:xnu-2782.50.1~1/RELEASE_X86_64
configure: error: cannot guess build type; you must specify one
```

The makefile log had some interesting things too:
`/Users/coen-hyde/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/extensions/x86_64-darwin-14/2.2.0-static/nokogiri-1.6.7.rc4/mkmf.log`

```
"clang -o conftest -I/Users/coen-hyde/.rbenv/versions/2.2.2/include/ruby-2.2.0/x86_64-darwin14 -I/Users/coen-hyde/.rbenv/versions/2.2.2/include/ruby-2.2.0/ruby/backward -I/Users/coen-hyde/.rbenv/versions/2.2.2/include/ruby-2.2.0 -I. -I/Users/coen-hyde/.rbenv/versions/2.2.2/include  -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -O3 -Wno-error=shorten-64-to-32  -pipe  conftest.c  -L. -L/Users/coen-hyde/.rbenv/versions/2.2.2/lib -L. -L/Users/coen-hyde/.rbenv/versions/2.2.2/lib  -fstack-protector     -lruby-static -framework CoreFoundation  -lpthread -ldl -lobjc  "
Undefined symbols for architecture x86_64:
  "_iconv", referenced from:
      _main in conftest-6da6c6.o
  "_iconv_open", referenced from:
      _main in conftest-6da6c6.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I've tried a lot of possible solutions for issues with iconv recommended in the issues and on stack overflow. None have works. Any thoughts on how to fix this?

**Comment by coen-hyde** [comment]
Some additional info:
```
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.10.5
BuildVersion:	14F1605
```

I do use Homebrew and have never installed MacPorts.

It looks like the #1423 might be the same issue.

**Comment by mojca** [comment]
Can you try to build 1.6.8.rc3?

**Comment by coen-hyde** [comment]
Yes same thing happens with 1.6.8.rc3. I probably should have mentioned that in the original issue. I just backtracked through the versions to determine where it broke.

**Comment by mojca** [comment]
Well, you mentioned issue #1423 and showed some log of missing `_libiconv`. The issue I reported (#1423) was magically fixed in 1.6.8.rc3 (I didn't test 1.6.8.rc2 to be honest).

But if this is still your main failure:
`````
configure: WARNING: if you wanted to set the --build type, don't use --host.
...
configure: error: cannot guess build type; you must specify one
`````
then it might make sense to figure out where the `--host` flag is coming from.

**Comment by flavorjones** [comment]
I believe if the installation instructions are followed from the [tutorial][http://www.nokogiri.org/tutorials/installing_nokogiri.html] that you'll be able to install successfully.
